oci/archive/reader.go

```diff
@@ -0,0 +1,90 @@
+package archive
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/image/v5/internal/tmpdir"
+	"github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/archive"
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type reader struct {
+	manifest *imgspecv1.Index
+	tempDirOCIRef
+}
+
+// CreateUntarTempDirReader creates the temp directory that keeps the untarred archive from src.
+// The caller should call .Close() on the returned object.
+func CreateUntarTempDirReader(ctx context.Context, src string, sys *types.SystemContext) (*reader, error) {
```

Comment:
Why not just a `NewReader`, to be consistent with docker/archive?

The `reader` type should probably be public (with all members private), so that the caller can declare a temporary variable or a parameter holding the returned value.